libs/portals/admin/regulations-admin/src/state/reducer.ts

```diff
@@ -0,0 +1,81 @@
+import { Reducer, useReducer } from 'react'
+import { Step } from '../types'
+import { LawChapter, MinistryList } from '@island.is/regulations'
+import { Action, DraftingState, RegDraftFormSimpleProps } from './types'
+import { produce, setAutoFreeze } from 'immer'
+import { DraftImpactId, RegulationDraft } from '@island.is/regulations/admin'
+import { useAuth } from '@island.is/auth/react'
+import { AdminPortalScope } from '@island.is/auth/scopes'
+import { derivedUpdates, validateState } from './validations'
+import { makeDraftForm, stepsAmending, stepsBase } from './makeFields'
+import { actionHandlers } from './actionHandlers'
+
+const draftingStateReducer: Reducer<DraftingState, Action> = (
+  state,
+  action,
+) => {
+  setAutoFreeze(false)
+  const newState = produce(state, (sDraft) =>
+    actionHandlers[action.type](
+      sDraft,
+      // @ts-expect-error  (Can't get this to work. FML)
```

Comment:
Get what to work?

Reply:
I have updated the comment. We could have a look at this at a later stage, but this is working, not sure why it's complaining.